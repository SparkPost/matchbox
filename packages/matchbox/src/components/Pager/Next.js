import React from 'react';
import { ArrowForward } from '@sparkpost/matchbox-icons';
import { Button } from '../Button';
import { ScreenReaderOnly } from '../ScreenReaderOnly';
import styles from './Pager.module.scss';

const Next = (props) => (
  <Button {...props} className={styles.Next}>
    <ArrowForward size={16} />

    <ScreenReaderOnly>Next</ScreenReaderOnly>
  </Button>
);

Next.displayName = 'Pager.Next';

export default Next;
