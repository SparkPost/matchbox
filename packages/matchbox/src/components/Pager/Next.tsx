import React from 'react';
import { ArrowForward } from '@sparkpost/matchbox-icons';
import { Button, PolymorphicButton } from '../Button';
import { ScreenReaderOnly } from '../ScreenReaderOnly';

const Next = React.forwardRef(function Next(props, userRef) {
  return (
    <Button
      size="small"
      width="600"
      {...props}
      ml={props.ml || props.marginRight || 200}
      ref={userRef}
    >
      <ArrowForward size={16} />
      <ScreenReaderOnly>Next</ScreenReaderOnly>
    </Button>
  );
}) as PolymorphicButton;

Next.displayName = 'Pager.Next';
export default Next;
