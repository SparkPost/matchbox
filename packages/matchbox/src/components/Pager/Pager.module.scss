@import '../../styles/config';

.Pager {
  display: inline-block;
  margin-bottom: spacing();
}

.Next {
  margin-left: spacing(small);
}

.Previous {
  margin-right: spacing(small);
}
