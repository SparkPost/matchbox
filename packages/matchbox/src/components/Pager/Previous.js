import React from 'react';
import { ArrowBack } from '@sparkpost/matchbox-icons';
import { Button } from '../Button';
import { ScreenReaderOnly } from '../ScreenReaderOnly';
import styles from './Pager.module.scss';

const Previous = (props) => (
  <Button {...props} className={styles.Previous}>
    <ArrowBack size={16} />

    <ScreenReaderOnly>Previous</ScreenReaderOnly>
  </Button>
);

Previous.displayName = 'Pager.Previous';

export default Previous;
