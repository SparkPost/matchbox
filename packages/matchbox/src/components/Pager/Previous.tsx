import React from 'react';
import { ArrowBack } from '@sparkpost/matchbox-icons';
import { Button } from '../Button';
import { ScreenReaderOnly } from '../ScreenReaderOnly';

type PreviousProps = React.ComponentProps<typeof Button>;

const Previous = (props: PreviousProps): JSX.Element => (
  <Button {...props} mr={props.mr || props.marginRight || 200}>
    <ArrowBack size={16} />
    <ScreenReaderOnly>Previous</ScreenReaderOnly>
  </Button>
);

Previous.displayName = 'Pager.Previous';

export default Previous;
