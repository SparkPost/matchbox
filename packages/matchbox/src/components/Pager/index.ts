export { default as Pager } from './Pager';
