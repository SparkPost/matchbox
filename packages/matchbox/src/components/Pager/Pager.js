import React, { Component } from 'react';
import Next from './Next';
import Previous from './Previous';
import styles from './Pager.module.scss';

class Pager extends Component {
  static displayName = 'Pager';

  static Next = Next
  static Previous = Previous

  render() {
    return (
      <div className={styles.Pager}>
        {this.props.children}
      </div>
    );
  }
}

export default Pager;
