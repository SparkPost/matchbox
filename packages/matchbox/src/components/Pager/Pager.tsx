import React from 'react';
import { margin, MarginProps } from 'styled-system';
import { pick } from '../../helpers/props';
import styled from 'styled-components';

import Next from './Next';
import Previous from './Previous';

const StyledPager = styled('div')`
  ${margin}
  display: inline-block;
`;

export type PagerProps = MarginProps & {
  'data-id'?: string;
  children?: React.ReactNode;
  id?: string;
};

function Pager(props: PagerProps): JSX.Element {
  const { children, id, 'data-id': dataId, ...rest } = props;
  const systemProps = pick(rest, margin.propNames);

  return (
    <StyledPager id={id} data-id={dataId} {...systemProps}>
      {children}
    </StyledPager>
  );
}

Pager.Next = Next;
Pager.Previous = Previous;
Pager.displayName = 'Pager';

export default Pager;
