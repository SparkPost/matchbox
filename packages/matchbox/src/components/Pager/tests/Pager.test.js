import React from 'react';
import { shallow } from 'enzyme';
import Pager from '../Pager';

describe('Pager', () => {
  it('renders pager with buttons', () => {
    const wrapper = shallow(
      <Pager>
        <Pager.Previous />
        <Pager.Next />
      </Pager>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
