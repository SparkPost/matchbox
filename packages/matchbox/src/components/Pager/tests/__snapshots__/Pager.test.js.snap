// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pager renders pager with buttons 1`] = `
<div
  className="Pager"
>
  <Pager.Previous />
  <Pager.Next />
</div>
`;
